code/datums/elements/limb_support.dm

```diff
@@ -0,0 +1,46 @@
+/datum/element/limb_support
+	element_flags = ELEMENT_BESPOKE
+	id_arg_index = 2
+
+	var/limbs_to_support = FULL_BODY
+
+/datum/element/limb_support/Attach(datum/target, limbflags)
+	. = ..()
+	if(limbflags)
+		limbs_to_support = limbflags
+
+	RegisterSignal(target, COMSIG_ITEM_EQUIPPED_TO_SLOT, .proc/equipped)
+	RegisterSignal(target, list(COMSIG_ITEM_EQUIPPED_NOT_IN_SLOT, COMSIG_ITEM_DROPPED), .proc/dropped)
+
+/datum/element/limb_support/proc/equipped(datum/source, mob/equipper, slot)
+	if(!ishuman(equipper))
+		return
+	RegisterSignal(equipper, COMSIG_HUMAN_LIMB_FRACTURED, .proc/fuckibrokemyleg)
+	fuckibrokemyleg(equipper, null, FALSE) // manually called when equipped to stabilize limbs
+
+/datum/element/limb_support/proc/dropped(datum/source, mob/equipper)
+	fuckibrokemyleg(equipper, null, TRUE) // manually called when dropped to remove stabilization of limbs
+	UnregisterSignal(equipper, COMSIG_HUMAN_LIMB_FRACTURED)
+
+/datum/element/limb_support/proc/fuckibrokemyleg(datum/owner, datum/limb/fractured_limb, dropped = FALSE)
```

Comment:
i love how you gave it a proper name but then kept this shitpost proc name i made :P